sw/device/lib/testing/usb_testutils.c

```diff
@@ -225,27 +416,24 @@ void usb_testutils_init(usb_testutils_ctx_t *ctx, bool pinflip,
   // All about polling...
   CHECK_DIF_OK(dif_usbdev_irq_disable_all(ctx->dev, NULL));
 
-  // Provide buffers for any reception
+  // Provide buffers for any packet reception
   CHECK_DIF_OK(dif_usbdev_fill_available_fifo(ctx->dev, ctx->buffer_pool));
 
-  dif_usbdev_endpoint_id_t endpoint = {
-      .number = 0,
-      .direction = 1,
-  };
-  CHECK_DIF_OK(
-      dif_usbdev_endpoint_enable(ctx->dev, endpoint, kDifToggleEnabled));
+  // Preemptively enable SETUP reception on endpoint zero for the
+  // Default Control Pipe; all other settings for that endpoint will be applied
+  // once the callback handlers are registered by a call to _endpoint_setup()
   CHECK_DIF_OK(
-      dif_usbdev_endpoint_stall_enable(ctx->dev, endpoint, kDifToggleDisabled));
+      dif_usbdev_endpoint_setup_enable(ctx->dev, 0, kDifToggleEnabled));
```

Comment:
For a later PR: Probably this should be absorbed into endpoint setup functions that take in the endpoint type. We don't have a way to specify control / bulk / interrupt / isochronous yet.

Reply:
Agreed, thanks. I have some incomplete code on another branch for setting up Isochronous endpoints, so I'll address this in that PR; it extends the usbdev_stream_test to exercise traffic to/from endpoints of different types concurrently and has been proved to some extent in t-l sim but needs more work on the physical host/FPGA setup at present.